blocklylib-core/src/main/java/com/google/blockly/model/Block.java

```diff
@@ -862,12 +867,32 @@ public Connection getParentConnection() {
      * @return The block connected to the output or previous {@link Connection}, if present.
      *         Otherwise null.
      */
+    @Nullable
     public Block getParentBlock() {
         Connection parentConnection = getParentConnection();
         return parentConnection == null ? null : parentConnection.getBlock();
     }
 
     /**
+     * Sets a event callback that will follow the block to any workspace (or workspace equivalent:
+     * toolbox or trash) it is attached to. This get called on all events.
+     * @param callback The block's callback or null
+     */
+    public void setEventCallback(@Nullable BlocklyController.EventsCallback callback) {
+        if (mEventCallback != null) {
+            mController.removeCallback(mEventCallback);
```

Comment:
Per discussion, added a wrapper that will automatically unregister if the Block gets garbage collected.